lib/pbench/cli/server/user_delete.py

```diff
@@ -0,0 +1,65 @@
+import click
+from pbench import BadConfig
+from pbench.server import PbenchServerConfig
+from pbench.cli.server import pass_cli_context
+from pbench.cli.server.options import common_options
+from pbench.common.logger import get_pbench_logger
+from pbench.server.database.models.users import User
+from pbench.server.database.database import Database
+
+_NAME_ = "pbench-user-delete"
+
+
+class UserDelete:
+    def __init__(self, context):
+        self.context = context
+
+    def execute(self):
+        config = PbenchServerConfig(self.context.config)
+
+        logger = get_pbench_logger(_NAME_, config)
+
+        # We're going to need the Postgres DB to track dataset state, so setup
+        # DB access.
+        Database.init_db(config, logger)
+
+        # Determine if the token belongs to the admin user
+        user = User.query(username=self.context.username)
+
+        # Check if the user has the right privileges
+        if user is None or not user.is_admin():
+            raise RuntimeError(f"Not authorized to delete user {self.context.username}")
+
+        # If the authorization passed delete the user from the given username
+        User.delete(username=self.context.user_to_delete)
+
+        click.echo(f"User {self.context.user_to_delete} deleted")
+
+
+@click.command()
+@common_options
+@click.option(
+    "--username",
+    required=True,
+    prompt=True,
+    help="pbench server account username (will prompt if unspecified)",
+)
```

Comment:
This seems pointless without any real authorization check; and a user logged in to the server can easily do it anyway so why make things more awkward by pretending?